### ISSUE TITLE
Exporting SVG not displaying dimensions

### ISSUE THREAD

**Issue by ezequielleonzybert** [LibreCAD/LibreCAD#1760]
https://github.com/LibreCAD/LibreCAD/assets/25140875/adbc7bba-43da-4b46-a6da-0e947742f482

When exporting the project as SVG the dimensions are not displayed in the svg file.

Quick workaround is to explode all the dimensions so they become simple lines.

**Comment by dxli** [comment]
The SVG maker in LibreCAD was created for CNC purposes, instead of a generic SVG generator.

It's possible to create a generic SVG version with better support for dimensions and texts.

**Comment by ezequielleonzybert** [comment]
> The SVG maker in LibreCAD was created for CNC purposes, instead of a generic SVG generator.
> 
> It's possible to create a generic SVG version with better support for dimensions and texts.

One could simply hide the layers that don't want to send to the CNC machine.

**Comment by dongxuli-concord-design** [comment]
Yes. I would pay more attention to preserving CNC support while adding new features. That means more development time, which is the actual limit here.

**Comment by ezequielleonzybert** [comment]
> Yes. I would pay more attention to preserving CNC support while adding new features. That means more development time, which is the actual limit here.

Do you mean that previous svg exporter had the functionality of exporting dimensions and then they exclude that function in order of preserving CNC support?

**Comment by dxli** [comment]
No. 

I mean:

1, the current maker only considers entities needed for CNC;
2, SVG supports dimensions! 
3, simply adding dimensions to the current maker is not desirable, for adding unwanted entities;
4, the proper way is to add a control to the current maker, or even in the form of a different maker with dimension, text, etc, support.

**Comment by dxli** [comment]
@ezequielleonzybert 

Do you want to work on this one? 

To expand SVG support is a great feature. Ideally, we can make svg a CAD format for LibreCAD.

**Comment by ezequielleonzybert** [comment]
> @ezequielleonzybert
> 
> Do you want to work on this one?
> 
> To expand SVG support is a great feature. Ideally, we can make svg a CAD format for LibreCAD.

I'd like to contribute. But I can see the project is already moving to version 3. Will this version become obsolete then?

**Comment by dongxuli-concord-design** [comment]
Anything v2 can be ported to v3. 

V3 is not actively being developed now.

I found v2 is still very useful, so I keep working on it.

We should also release another stable release for v2 soon.
